### PR TITLE
[MCI] Support for handling MCI objects

### PR DESCRIPTION
**Pull request by swathinsankaran**
This PR adds support for handling MCI objects in amko.

### CODE REVIEW

File: gslb/ingestion/event_handlers.go [hunk 1]
```diff
@@ -696,3 +696,186 @@ func AddHostRuleEventHandler(numWorkers uint32, c *GSLBMemberController) cache.R
...
+				continue
+			}
+			// Else, delete this ingressHost from accepted list and add the newIhm to the
+			// rejected store, and add a delete key for this ingressHost to the queue
+			AddOrUpdateMultiClusterIngressStore(rejectedStore, newMCIhm, c.name)
+			DeleteFromMultiClusterIngressStore(acceptedStore, newMCIhm, c.name)
+
+			fetchedIngHost := fetchedObj.(k8sobjects.IngressHostMeta)
+			// Add a DELETE key for this ingHost
+			publishKeyToGraphLayer(numWorkers, gslbutils.IngressType, fetchedIngHost.Cluster,
```
Comment (RiyaGaniga-avi) [inline gslb/ingestion/event_handlers.go:771]: Here the type has to be `gslbutils.MCIType` instead of `gslbutils.IngressType`

Comment (swathinsankaran) [reply gslb/ingestion/event_handlers.go]: I have corrected the object type passed to the graph layer.

**Review by RiyaGaniga-avi — APPROVED** [review]

File: helm/amko/templates/statefulset.yaml [hunk 2]
```diff
@@ -57,6 +57,10 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.name
+          {{ if .Values.multiClusterIngress.enable }}
+          - name: MCI_ENABLED
```
Comment (monotosh-avi) [inline helm/amko/templates/statefulset.yaml:61]: Are we using MCI_ENABLED anywhere ?

File: gslb/k8sobjects/multicluster_ingress_objects.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+		// won't be processed, this is just to maintain the ingress information as part
+		// of in-memory map.
+		gslbutils.Errf("ns: %s, ingress: %s, msg: skipping ingress because of error: %v",
+			mci.Namespace, mci.Name, err)
+	}
+	if (controllerUUID == "" || len(vsUUIDs) == 0) && !syncVIPsOnly {
+		gslbutils.Logf("cluster: %s, ns: %s, ingress: %s, msg: skipping ingress because controller UUID absent in annotations",
+			cname, mci.Namespace, mci.Name)
+	}
+	for _, hip := range hostIPList {
```
Comment (monotosh-avi) [inline gslb/k8sobjects/multicluster_ingress_objects.go:78]: I think one MCI object can only have one host. In that case hostIPList would have only one value. Probably we don;t need a list here. Though its fine this way too.

File: helm/amko/templates/statefulset.yaml (hunk 2 above)
Comment (swathinsankaran) [reply helm/amko/templates/statefulset.yaml:61]: yes, we are using the flag while initializing the MCI informers.

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**